File: Parth Mathur/57_PascalTriangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        
        if(numRows == 0)
        {
            return new ArrayList<>();
        }
        
        List<List<Integer>> ans = new ArrayList();
        
        for(int i=1; i<=numRows; i++)
        {
            List<Integer> smallAns = new ArrayList();
            for(int j=0; j<i ; j++)
            {
                if(j==0 || j==i-1)
                {
                    smallAns.add(1);
                }
                else
                {
                    smallAns.add(ans.get(i-2).get(j) +  ans.get(i-2).get(j-1));
                }
                
            }
            ans.add(smallAns);
        }
        
        return ans;
        
    }
}

File: Dhairyashil-Shinde/57. Pascal Triangle.cpp
class Solution {
public:
    
    vector<vector<int>> generate(int numRows) {
        
        int n = numRows;
        
        vector<vector<int>> ans(n);             
    
        for(int i = 0; i < n; i++) {
        
            ans[i] = vector<int>(i+1,1);        
            
            for(int j = 1; j < i; j++) {
            
                 ans[i][j] = ans[i - 1][j] + ans[i - 1][j - 1];            
            
            }
        }
        
        return ans;
    }
};
